sn_logging/src/metrics.rs

```diff
@@ -0,0 +1,146 @@
+}
+
+#[derive(Debug, Serialize)]
+#[allow(dead_code)]
+struct ProcessMetrics {
+    // Percentage of CPU used by the process
+    cpu_usage_percent: f32,
+    // RAM used in MBytes
+    memory_used_mb: f32,
+    // Bytes read from disk during UPDATE_INTERVAL
+    bytes_read: u64,
+    // Bytes written to disk during UPDATE_INTERVAL
+    bytes_written: u64,
+    // The total MBytes read from disk by the process
+    total_mb_read: f32,
+    // The total MBytes written to disk by the process
+    total_mb_written: f32,
+}
+
+// Obtains the system metrics every UPDATE_INTERVAL and logs it.
+// The function should be spawned as a task and should be re-run if our main process is restarted.
+pub async fn init_metrics(pid: u32) {
+    let mut sys = System::new_all();
+    let pid = Pid::from_u32(pid);
+    let default_interface = default_net::get_default_interface();
+
+    loop {
+        refresh_metrics(&mut sys, pid);
+
+        let process = match sys.process(pid) {
+            Some(safenode) => {
+                let disk_usage = safenode.disk_usage();
+                let process = ProcessMetrics {
+                    cpu_usage_percent: safenode.cpu_usage(),
+                    memory_used_mb: safenode.memory() as f32 / TO_MB,
+                    bytes_read: disk_usage.read_bytes,
+                    bytes_written: disk_usage.written_bytes,
+                    total_mb_read: disk_usage.total_read_bytes as f32 / TO_MB,
+                    total_mb_written: disk_usage.total_written_bytes as f32 / TO_MB,
+                };
+                Some(process)
+            }
+            None => {
+                // safenode with the provided Pid not found
+                None
+            }
+        };
+
+        let network = if let Ok(default_interface) = &default_interface {
+            if let Some((interface_name, network_stat)) = sys
+                .networks()
+                .into_iter()
+                .find(|&(interface, _)| interface == &default_interface.name)
+            {
+                let network = NetworkMetrics {
+                    interface_name: interface_name.clone(),
+                    bytes_received: network_stat.received(),
+                    bytes_transmitted: network_stat.transmitted(),
+                    total_mb_received: network_stat.total_received() as f32 / TO_MB,
+                    total_mb_transmitted: network_stat.total_transmitted() as f32 / TO_MB,
+                };
+                Some(network)
+            } else {
+                // Could not get stats for the default interface
+                None
+            }
+        } else {
+            // Could not obtain the default network interface");
+            None
+        };
+
+        let cpu_stat = sys.global_cpu_info();
+        let metrics = Metrics {
+            physical_cpu_threads: sys.cpus().len(),
+            cpu_usage_percent: cpu_stat.cpu_usage(),
+            total_memory_mb: sys.total_memory() as f32 / TO_MB,
```

Comment:
this is about the whole system usage?
I think we don't care much about it?

Reply:
Users will though and if it is included I will add it to `vdash`.

Reply:
> this is about the whole system usage?
Yes, I've renamed them to reflect that.
Might be interesting to see how the binaries run on low-end devices?

Reply:
This is the final JSON that is logged,
```
{"physical_cpu_threads":12,"system_cpu_usage_percent":8.87408,"system_total_memory_mb":33517.777,"system_memory_used_mb":14960.186,"system_memory_usage_percent":44.633587,"network":{"interface_name":"enp0s31f6","bytes_received":1350,"bytes_transmitted":53303,"total_mb_received":2376.993,"total_mb_transmitted":749.26776},"process":{"cpu_usage_percent":4.6299543,"memory_used_mb":38.64576,"bytes_read":0,"bytes_written":16384,"total_mb_read":0.0,"total_mb_written":4.251648}}
```